### ISSUE TITLE
FL: Links / interactive elements within a scrollable div should have tab index + Provider search test flaky AXE error

### ISSUE THREAD

**Issue by jilladams** [department-of-veterans-affairs/va.gov-cms#19189]
## User Story or Problem Statement
In the Facility Locator, we have an accessibility issue that is also associated with a flaky Axe test. 

## Description or Additional Context 
WCAG has a rule that any interactive element within a scrollable div must have its own tab index, in order to allow users to tab to it. 
In the Facility Locator, our search results have interactive links that must be tabbable. In some cases, links within a search result card are not correctly being assigned a tabindex. (Is this true / did I get it right?)

We also have an automated test in vets-website that uses Axe to check tab indexes, and sometimes fails: `src/applications/facility-locator/tests/e2e/providerSearch.cypress.spec.js`. This test has turned up in the flaky test report (9/3), and when it fails on a PR, it blocks merge of that PR. 

Example URL where a search result with an interactive link within a scrollable div should have a tabindex: (?)
https://dev.va.gov/find-locations/?address=Pasadena%2C%20CA&context=Pasadena%2C%20California%2C%20United%20States&facilityType=provider&page=1&serviceType=122300000X&latitude=34.145775&longitude=-118.150475&radius=10&bounds%5B%5D=-118.900475&bounds%5B%5D=33.395775&bounds%5B%5D=-117.400475&bounds%5B%5D=34.895775

## Acceptance Criteria
- [ ] Test runs and passes reliably
- [ ] Requires accessibility review

**Comment by jilladams** [comment]
This came up on https://github.com/department-of-veterans-affairs/vets-website/pull/31801, where eventually the flaky test didn't flake / PR could merge but: it was annoying, so we decided to ticket since this could happen to any vets-website PR where the test fails.
